program/src/error.rs

```diff
@@ -0,0 +1,78 @@
+//! Error types
+
+use {
+    num_derive::FromPrimitive,
+    solana_program::{
+        decode_error::DecodeError,
+        msg,
+        program_error::{PrintProgramError, ProgramError},
+    },
+    std::error::Error,
+    thiserror::Error,
+};
+
+/// Errors that may be returned by the Token Wrap program.
+#[derive(Clone, Debug, Eq, Error, PartialEq, FromPrimitive)]
+pub enum TokenWrapError {
+    // 0
+    /// Wrapped mint account address does not match expected PDA
+    #[error("Wrapped mint account address does not match expected PDA")]
+    WrappedMintMismatch,
+    /// Wrapped backpointer account address does not match expected PDA
+    #[error("Wrapped backpointer account address does not match expected PDA")]
+    BackpointerMismatch,
+    /// Wrap amount should be positive
+    #[error("Wrap amount should be positive")]
+    ZeroWrapAmount,
+    /// Wrapped mint authority does not match expected PDA
+    #[error("Wrapped mint authority does not match expected PDA")]
+    MintAuthorityMismatch,
+    /// Unwrapped escrow token owner is not set to token-wrap program
+    #[error("Unwrapped escrow token owner is not set to token-wrap program")]
+    EscrowOwnerMismatch,
```

Comment:
nit: the escrow owner is a PDA, and more specifically, the wrapped mint authority, so probably clearer to say `Unwrapped escrow token owner is not set to expected PDA`